src/app/clusters/concentration-measurement-server/concentration-measurement-cluster-objects.h

```diff
@@ -0,0 +1,268 @@
+#include <app/CommandHandlerInterface.h>
+#include <app/util/util.h>
+#include <utility>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ConcentrationMeasurement {
+
+static constexpr std::array<ClusterId, 10> AliasedClusters = { CarbonDioxideConcentrationMeasurement::Id,
+                                                               CarbonMonoxideConcentrationMeasurement::Id,
+                                                               NitrogenDioxideConcentrationMeasurement::Id,
+                                                               Pm1ConcentrationMeasurement::Id,
+                                                               Pm10ConcentrationMeasurement::Id,
+                                                               Pm25ConcentrationMeasurement::Id,
+                                                               RadonConcentrationMeasurement::Id,
+                                                               TotalVolatileOrganicCompoundsConcentrationMeasurement::Id,
+                                                               OzoneConcentrationMeasurement::Id,
+                                                               FormaldehydeConcentrationMeasurement::Id };
+
+enum class LevelValueEnum : uint8_t
+{
+    kUnknown  = 0x00,
+    kLow      = 0x01,
+    kMedium   = 0x02,
+    kHigh     = 0x03,
+    kCritical = 0x04,
+    // All received enum values that are not listed above will be mapped
+    // to kUnknownEnumValue. This is a helper enum value that should only
+    // be used by code to process how it handles receiving and unknown
+    // enum value. This specific should never be transmitted.
+    kUnknownEnumValue = 5,
+};
+
+enum class MeasurementMediumEnum : uint8_t
+{
+    kAir   = 0x00,
+    kWater = 0x01,
+    kSoil  = 0x02,
+    // All received enum values that are not listed above will be mapped
+    // to kUnknownEnumValue. This is a helper enum value that should only
+    // be used by code to process how it handles receiving and unknown
+    // enum value. This specific should never be transmitted.
+    kUnknownEnumValue = 3,
+};
+
+enum class MeasurementUnitEnum : uint8_t
+{
+    kPpm  = 0x00,
+    kPpb  = 0x01,
+    kPpt  = 0x02,
+    kMgm3 = 0x03,
+    kUgm3 = 0x04,
+    kNgm3 = 0x05,
+    kPm3  = 0x06,
+    kBqm3 = 0x07,
+    // All received enum values that are not listed above will be mapped
+    // to kUnknownEnumValue. This is a helper enum value that should only
+    // be used by code to process how it handles receiving and unknown
+    // enum value. This specific should never be transmitted.
+    kUnknownEnumValue = 8,
+};
+
+enum class Feature : uint32_t
+{
+    kNumericMeasurement = 0x1,
+    kLevelIndication    = 0x2,
+    kMediumLevel        = 0x4,
+    kCriticalLevel      = 0x8,
+    kPeakMeasurement    = 0x10,
+    kAverageMeasurement = 0x20,
+};
+
+namespace Attributes {
+
+namespace MeasuredValue {
+static constexpr AttributeId Id = 0x00000000;
+struct TypeInfo
```

Comment:
Are any of these TypeInfo bits used anywhere?  I would expect them to be used on the client, and totally unused on the server.  Unless something uses TypeInfo::Type for declaring variables, but nothing seems to.